org.osgi.test.junit4/src/main/java/org/osgi/test/junit4/BundleContextRule.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) OSGi Alliance (2019). All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.osgi.test.junit4;
+
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.test.common.CloseableBundleContext;
+
+/**
+ * A JUnit 4 Rule to get the OSGi {@link BundleContext} of the test bundle.
+ * <p>
+ * The {@link BundleContext} implementation provided by this rule will
+ * automatically clean up all service registrations, bundle, service and
+ * framework listeners, as well as installed bundles left behind.
+ * <p>
+ * Example:
+ *
+ * <pre>
+ * &#64;Rule
+ * public BundleContextRule rule = new BundleContextRule();
+ *
+ * &#64;Test
+ * public void aTest() {
+ * 	Bundle bundle = rule.get()
+ * 		.getBundle();
+ * }
+ * </pre>
+ */
+public class BundleContextRule implements TestRule {
+
+	private CloseableBundleContext bundleContext;
```

Comment:
We should not use this type here. We can test in the finally if the context implements AutoClosable and then cast and call close.

Reply:
done!